htmltools/_core.py

```diff
@@ -358,17 +361,28 @@ def show(self, renderer: Literal["auto", "ipython", "browser"] = "auto") -> obje
         """
         _tag_show(self, renderer)
 
-    def __str__(self) -> str:
-        return str(self.get_html_string())
-
     def __eq__(self, other: Any) -> bool:
         return _equals_impl(self, other)
 
+    def __str__(self) -> str:
+        rendered = self.render()
+        res = rendered["html"]
```

Comment:
This should be a function

Reply:
Maybe add a comment that this is for Quarto integration

Reply:
Done in abab13d.